src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1411,14 +1432,21 @@ private void visitTemplateTypes(ObjectType type, Set<String> alreadyEmittedTempl
           // Some template variables can be already defined at the class definition.
           // Closure and TypeScript disagree in that case, in closure redeclaring a class template
           // variable at a method does nothing, but in Typescript it introduces a new variable.
-          // To perserve the sementics from closure we skip emitting redeclared variables.
+          // To preserve the semantics from closure we skip emitting redeclared variables.
           if (alreadyEmittedTemplateType.contains(displayName)) {
             continue;
           }
 
           if (displayName.contains("IObject#")) {
             displayName = normalizeIObjectTemplateName(type, displayName);
           }
+          // When we emit partial programs, we cannot differentiate whether Foo
```

Comment:
This is a good fix, but I think it might be even more generic. AFAIU generally the semantics of Closure are that you can leave out generic type arguments, and it'll simply assume an unknown type (~= any) in its place. So this is correct under all circumstances, not just for partialInput, isn't it?

That'd allow us to drop the logic that inserts `<any>` for missing template parameters in the other place.

Reply:
Actually, thinking about it a bit more, I'm no longer that sure. Using a class but forgetting its generic argument and then getting any is a bit sad - you lose a lot of type safety, and it's very easy to do so unintentionally.

I wonder if we can default to `{}` (empty type) instead of `any`?

Reply:
Ack, I think this is fine.